### ISSUE TITLE
Reporting a song as needing a quality check

### ISSUE THREAD

**Issue by codeheroics** [karakuri-js/karakuri-mobile#49]
As talked about in https://github.com/karakuri-js/karakuri-mobile/issues/46 , we need some way to report songs for bad subtitles or delayed sound or video.

Basically, we could have a :warning: icon on the songs list / (currently non-existing) song screen.
I see two possibilities : 
1. Touching the icon sends the report
1. Touching the icon sends the user to a "Reports" screen, a form, on which he can choose what the problem (radio buttons to choose from "lyrics error" / "video error" / other) is and / or add a comment, then send it.

We'll need a karakuri-server issue to talk about how we handle the reports.

**Comment by codeheroics** [comment]
I like option 2 better, even though it makes for a slower experience

**Comment by Fenntasy** [comment]
Yeah I don't like the specialized thing and if I must, I prefer option 1
because just launching the video will clearly tell where is the problem.

We may need someone to make the split on the issue, @zidrune ?

**Comment by codeheroics** [comment]
I disagree, launching the video won't tell where the problem is if it's a subtle issue (like a typo in the subtitles), the whole video + subtitles would need to be reviewed, which could be quite a pain. The specific screen could allow us to specifically point at what's wrong... while also letting you be done in two taps if you don't want to bother adding a comment.

But yeah, I'm good with letting @Zidrune decide :)

**Comment by Fenntasy** [comment]
Yeah, what I had in mind was playlists a la YouTube. So we could have
different playlists for each problems and then have one system to manage
for all these things.

**Comment by codeheroics** [comment]
We are already creating m3u files server-side so I guess creating a playlists of problematic songs would be quite easy.

What I had in mind when talking about commenting on specific problems was more "yeah there's a problem with this line" thing, and less about generic issues (because yes, I think 90% of the issues will be "those lyrics are wrong/weird/bad/inexistent" and the last 10% will be about problematic videos)

Thinking about it, I'm not really sure if this kind of specific reporting would really be doable while singing... I'll let the person who will most likely handle most of these reports decide how we should do this :)

**Comment by Zidrune** [comment]
I like when I get to decide! :D
I'm for a full report system in two clicks : Report button next to the name of the current playing song > Choose from a list of most frequent reasons to reports (i'll think about the exact content of that list after SB).

In a perfect world, you'll also save the username reporting so I can bother him when I don't find what's wrong with my lyrics! ^^

**Comment by Fenntasy** [comment]
Allright, I concede.

What about storage of these then? A file? An email?

The email allows for some kind of todo list and avoid the storage but add a configuration info to know where to send them.
The file would be more crude to create and manage.

Other ideas?

**Comment by codeheroics** [comment]
I plan on working on this, but after re-reading the whole thread I don't think we ever agreed on how to implement this…

Since it'll probably be quicker to do for starters, I plan on creating the report screen (as defined in https://github.com/karakuri-js/karakuri), and to create files on the server when a report is made. If we're motivated enough, we'll add stuff like email or discord later ^^'
